crates/miden-objects/src/account/builder/mod.rs

```diff
@@ -171,6 +182,13 @@ impl AccountBuilder {
             .id_anchor
             .ok_or_else(|| AccountError::BuildError("anchor must be set".into(), None))?;
 
+        if self.storage_mode.is_private() && self.network_flag.is_enabled() {
```

Comment:
nit: I would probably use `!self.storage_mode.is_public()` instead of `self.storage_mode.is_private()`.